Cartesian speed limitation in PJTC [Doc update] (#296)
PilzDE/pilz_robots: prbt_hardware_support/README.md

#### Table of Contents
- [Introduction](#introduction)
- [Supported DIN EN ISO 10218-1 safety features](#supported-din-en-iso-10218-1-safety-features)
  - [Needed/supported hardware](#neededsupported-hardware) 
  - [Definitions](#definitions)
  - [Safe stop 1](#safe-stop-1-ss1)
  - [Brake test](#brake-test)
  - [Operation modes](#operation-modes)
  - [Speed monitoring](#speed-monitoring)
- [Overview system components](#overview-system-components)

# Introduction

> The `prbt_hardware_support` package provides support for the Pilz hardware

Reviewer: There is a video on this: https://vimeo.com/378682930 ;-)

PNOZmulti and PSS 4000. A number of safety features are provided which are
essential for a DIN EN ISO 10218-1 certifiable robot system (for more
information see section about 
[supported DIN EN ISO 10218-1 safety features](#supported-din-en-iso-10218-1-safety-features)).  
  
Due to the fact, that the communication between the Pilz safety controllers 
and ROS is based on the Modbus communication protocol, the package also 
contains C++ files providing ROS support for the Modbus communication protocol.
  
In the [system overview section](#overview-system-components),
you can find a component diagram showing the overall architecture of our
system. The component diagram shows all our nodes and the connections 
between them.

**Please note:**  
The launch files included in the `prbt_hardware_support` package don't need to
be called directly by the user. They already are included in our top-level
launch file (for more information on how-to run the PRBT robot with ROS see the 

> [pilz_robots-README](https://github.com/PilzDE/pilz_robots#running-on-the-real-robot).

Reviewer: we could reference https://github.com/PilzDE/pilz_templates/ here

# Supported DIN EN ISO 10218-1 safety features
The prbt_hardware_support package provides support for a number of safety
features which are essential for a DIN EN ISO 10218-1 certifiable robot system.
  
**Please note:**  

> The DIN EN ISO 10218-1 support is WORK IN PROGRESS!!!  

Reviewer: ```suggestion
The DIN EN ISO 10218-1 support is currently **WORK IN PROGRESS**  
```

 
## Needed/supported hardware
In order for the safety features to work, one needs special hardware

> components supporting special safety feature.

Reviewer: ```suggestion
components supporting the required safety features.
```

Currently, we test all our features against the following 
hardware setup:

> - Robot: Manipulator module PRBT

Reviewer: ```suggestion
- Robot: Manipulator module PRBT 6
```

> - Safety controller: PSS 4000 (with special PSS 4000 program)

Reviewer: ```suggestion
- Safety controller: PSS 4000 (with a dedicated program)
```

- Operating mode selector switch: PITmode
- Enabling switch: PITenable

> - Pushbutton unit (with emergency stop): PITgatebox

Reviewer: this can have various configurations

**Please note:**  
Not all safety features in this section are supported by both PILZ safety 

> controllers! Currently, the PNOZmulti does not support the brake test feature 

Reviewer: this is the first time, PNOZmulti is mentioned. Either introduce it as (legacy, unsupported) option or leave it out altogether.

See DIN EN ISO 10218-1 for more details or contact us: ros@pilz.de

**Please note:**  
In operation mode T1 the robot can be moved as usual. 
However, if an attempt to exceed the speed limit of 250 mm/s in T1 is detected,

> the prevailing motion is aborted and a controlled stop is performed.

Reviewer: ```suggestion
the current motion is aborted and a controlled stop is performed.
```
